src/main.rs

```diff
@@ -91,44 +140,82 @@ async fn redirect(State(shared_state): State<AppState>, data_string: String) ->
         Ok(uuid) => uuid,
         Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
     };
-    println!("{}", uuid);
+
     axum::response::Redirect::to(format!("/{}", uuid).as_str()).into_response()
 }
 
-fn migrate(data_string: String) -> Result<String, anyhow::Error> {
-    let tmp_dir: tempfile::TempDir = tempdir()?;
-
-    let input_tmpfile: tempfile::NamedTempFile = tempfile::Builder::new()
-        .suffix(".xml")
-        .tempfile_in(tmp_dir.path())?;
+async fn redirect(State(shared_state): State<AppState>, data_string: String) -> Response {
+    let database: tokio::sync::MutexGuard<'_, Connection> = shared_state.database.lock().await;
+    let data_arr: Vec<File> = vec![File {
+        file_content: data_string,
+        file_name: "file.xml".to_string(), //was mach ich mit dir
+        file_type: FileType::from_str("text/plain").unwrap()//remove unwrap
+    }];
+    let path = match migrate(data_arr) {
+        Ok(path) => path,
+        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+    };
 
-    std::fs::write(&input_tmpfile, data_string.as_bytes())?;
+    let uuid = match create_and_add_row(path, &database) {
+        Ok(uuid) => uuid,
+        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+    };
+    println!("{}", uuid);
+    axum::response::Redirect::to(format!("/{}", uuid).as_str()).into_response()
+}
 
+fn migrate(data_arr: Vec<File>) -> Result<String, anyhow::Error> {
     let output_tmpfile: tempfile::NamedTempFile = tempfile::Builder::new()
         .prefix("nm-migrated.")
         .suffix(".tar")
         .keep(true)
         .tempfile()?;
+
     let output_path_str: &str = output_tmpfile.path().to_str().unwrap();
 
-    let input_path_filename: &str = input_tmpfile
-        .path()
-        .file_name()
-        .ok_or(anyhow::anyhow!("Invalid filename"))?
-        .to_str()
-        .ok_or(anyhow::anyhow!("Invalid filename"))?;
-
-    let arguments_str = format!("run --rm -v {}:/migration-tmpdir:z {} bash -c 
-        \"migrate-wicked migrate -c /migration-tmpdir/{} && cp -r /etc/NetworkManager/system-connections /migration-tmpdir/NM-migrated\"", 
-        tmp_dir.path().display(),
-        REGISTRY_URL,
-        input_path_filename
-    );
-
-    let command_output = Command::new("podman")
+    let migration_target_tmpdir: tempfile::TempDir = tempdir()?;
+
+    for file in &data_arr {
+        let input_file_path = migration_target_tmpdir.path()
+            .join(&file.file_name);
+        println!("input_file_path: {}", input_file_path.to_string_lossy());
+        fs::File::create_new(&input_file_path).unwrap();
+        std::fs::write(&input_file_path, file.file_content.as_bytes())?;
+    }
+
+    let arguments_str = if data_arr[0].file_name.contains("ifcfg") {
```

Comment:
Since there is now a better way of identifying the file type you should use it here